### PR TITLE
Output Compiled Binaries to Same Directory as C++ Input File & Change Help Text

### PR DESCRIPTION
**Pull request by lyndonhenry**
- binary executables are now put in the same directory as their C++ source by souffle-compile.in
- help text has been changed in main.cpp for -c, -g, -o options to achieve better clarity and readability
- .souffle-sh script updated to avoid clang-format versioning issues, also now gets ubuntu packages directly from the README so as not to require maintainence

### CODE REVIEW

**Comment by mmcgr** [comment]
Should we put binaries in the same directory as the source? To me it seems more appropriate to put them in the current directory, and if there is a use case for putting compiled files elsewhere then do so with a new -o <location> flag.

**Comment by lyndonhenry** [comment]
Sounds good, I'll revert to the default behavior for no arguments and add a new -o flag.

**Comment by lyndonhenry** [comment]
Wait, no, sorry, I remember my line of reasoning now. 

The purpose of the souffle-compile script is to be used by Soufflé internally to compile its executables, unless the user calls this script directly (which I assume not to be a common use case, although this is an entirely anecdotal claim) then they have no interaction with it. Thus its output location should reflect the output location given to Souffle itself, with one of the -c, -o, or -g flags. I feel this is more consistent as these flags determine the location of the generated C++ code.

**Comment by lyndonhenry** [comment]
The alternative is, of course, for the script itself to have a -o flag, and for Soufflé to call it with this -o flag and the output directory of the C++ file. This is, I think, probably the best solution.

File: .travis/install.sh [hunk 1]
```diff
@@ -18,7 +18,6 @@ then
   if [ "$TEST_FORMAT" == 1 ]
   then
     sudo apt-get -y install clang-format-3.8
-
```
Comment (mmcgr) [inline .travis/install.sh]: This line is to visually separate the install step from the 'hack around travis limitations' step. Could you put it back?

(If we were to remove it, then we should also be removing similar section separating lines further down.)

Comment (lyndonhenry) [inline .travis/install.sh]: Ahh, yes, sorry about that.

**Pull request merged**